schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/balance/DefaultBalanceService.java

```diff
@@ -161,8 +164,19 @@ public void payment(Payment payment) {
 
     @Override
     public void applyTransfer(Transfer transfer) {
-        applyTransferTransaction(transfer, true);
-        applyTransferTransaction(transfer, false);
+        try {
+            applyTransferTransaction(transfer, true);
+        } catch (ModelException e) {
+            // TODO: This should not happen but if it ever does schedule a task to retry
+            log.error("Failed to make debit", e);
+        }
+
+        try {
+            applyTransferTransaction(transfer, false);
+        } catch (ModelException e) {
+            // TODO: This should not happen but if it ever does schedule a task to retry
+            log.error("Failed to make credit", e);
```

Comment:
I think its possible that a datastore contention could occur when doing the transfer, for example due to account balance move in Jasify account under heavy traffic. If that were to happen than scheduling a task to retry later would resolve the issue. The todo is just a reminder for the future to look further into this.